doc/tests/check-nix-code-blocks.nix

```diff
@@ -0,0 +1,27 @@
+{
+  runCommand,
+  markdown-code-runner,
+  nixfmt-rfc-style,
+}:
+
+runCommand "manual_check-nix-code-blocks"
+  {
+    nativeBuildInputs = [
+      markdown-code-runner
+      nixfmt-rfc-style
+    ];
+  }
+  ''
+    set +e
+
+    mdcr --check --config ${./mdcr-config.toml} ${./..}
+
+    if [ $? -ne 0 ]; then
+      echo -e "Error: mdcr command failed. Please make sure the Nix code snippets in Markdown files are correctly formatted.\n"
```

Comment:
Why `echo -e` and not `cat << EOF`? It's kinda hard to read and edit that way.

Reply:
Fixed.